src/react-native-cool-modals/NativeStackView.js

```diff
@@ -59,6 +51,43 @@ function ScreenView({ colors, descriptors, navigation, route, state }) {
     topOffset,
     transitionDuration,
   } = options;
+  const prevLongFormHeight = usePrevious(longFormHeight);
+  const prevShortFormHeight = usePrevious(shortFormHeight);
+
+  const context = useMemo(
+    () => ({
+      jumpToLong: () => {
```

Comment:
What would you say for an alternative solution?
`jumpToLong` and `jumpToShort` can accept an additional optional argument that will be a new height of the form in the new state? It will modify the navigation option (if needed) and at the same time snap to a new position (if needed). Then we will assure that each action is intentional from the user-side.